### PR TITLE
Adding support for the additional fixed sample mask in MSL

### PR DESCRIPTION
**Pull request by ponitka**
This includes updating a SPIRV-cross dependency.

### CODE REVIEW

File: spvc/src/main.cc [hunk 1]
```diff
@@ -294,6 +295,17 @@ int main(int argc, char** argv) {
         return 1;
       }
       msl_discrete_descriptor.push_back(descriptor_num);
+    } else if (args.starts_with("--msl-additional-fixed-sample-mask=")) {
+      string_piece descriptor_str;
```
Comment (Kangz) [inline spvc/src/main.cc:299]: sample_mask_str?

Comment (ponitka) [reply spvc/src/main.cc]: Done

**Comment by Kangz** [comment]
```
[496/586] Building CXX object spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o
FAILED: spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o
/usr/bin/clang++   -I../libshaderc/include -I../third_party/spirv-tools/include -I../libshaderc_spvc/include -I../third_party/spirv-tools -Ithird_party/spirv-tools -I../libshaderc_util/include -Wimplicit-fallthrough -g -isysroot /Applications/Xcode_11.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14   -Wextra-semi -Wall -Werror -fvisibility=hidden -fPIC -std=c++11 -std=gnu++11 -MD -MT spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o -MF spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o.d -o spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o -c ../spvc/src/main.cc
../spvc/src/main.cc:298:16: error: use of undeclared identifier 'args'; did you mean 'arg'?
    } else if (args.starts_with("--msl-additional-fixed-sample-mask=")) {
               ^~~~
               arg
../spvc/src/main.cc:188:24: note: 'arg' declared here
    const string_piece arg = argv[i];
```

**Comment by ponitka** [comment]
> ```
> [496/586] Building CXX object spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o
> FAILED: spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o
> /usr/bin/clang++   -I../libshaderc/include -I../third_party/spirv-tools/include -I../libshaderc_spvc/include -I../third_party/spirv-tools -Ithird_party/spirv-tools -I../libshaderc_util/include -Wimplicit-fallthrough -g -isysroot /Applications/Xcode_11.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14   -Wextra-semi -Wall -Werror -fvisibility=hidden -fPIC -std=c++11 -std=gnu++11 -MD -MT spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o -MF spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o.d -o spvc/CMakeFiles/spvc_exe.dir/src/main.cc.o -c ../spvc/src/main.cc
> ../spvc/src/main.cc:298:16: error: use of undeclared identifier 'args'; did you mean 'arg'?
>     } else if (args.starts_with("--msl-additional-fixed-sample-mask=")) {
>                ^~~~
>                arg
> ../spvc/src/main.cc:188:24: note: 'arg' declared here
>     const string_piece arg = argv[i];
> ```

Fixed

**Comment by zoddicus** [comment]
Looks like the test expectations changed because you were rolling SPIRV-Cross. I will do the weekly DEPS roll, which will fix that, then you can rebase & remove your DEPS change. I will let you know when that is done.

**Comment by zoddicus** [comment]
I landed https://github.com/google/shaderc/pull/1121, so you should be able to rebase your patch, remove the DEPS change, and upload a new version that will pass the bots.

**Comment by ponitka** [comment]
Thanks. I rebased the patch.

**Comment by Kangz** [comment]
@austinEng FYI

**Pull request merged**